### DISCUSSION TITLE
Workflow storing different presentations on GitHub

### DISCUSSION THREAD

**Discussion by jensbri** [quarto-dev/quarto-cli#6913] [Q&A]
### Description

Kia ora/hello from New Zealand.

I am integrating Quarto and especially revealJS more and more into my daily life/replacing PowerPoint of Google Slides.
Starting out, I created one GitHub repo per slide deck.
That proves to be somewhat cumbersome, as extensions have to be installed, the GitHub render action needs to be configured, and it is far from a one-click solution.
Still, the underlying steps seem to be relatively standardised. 
I was wondering if an extension or similar existed/some best practices to be shared around how to store/share/hide several slide decks?

Cheers/Ngā mihi
Jens

**Comment by andrewpbray** [comment]
I recommend putting all of your slide decks into one repo as part of a Quarto Project. If you use a default project, that will get you the ability to install your extensions once at the project level and be available to all slide decks that call on that extension. If the project type is a website, then you can put together a simple boilerplate website using `quarto create-project --type website`, link to the slide decks, then publish using gh-pages or Quarto pub.

Here's a repo that demonstrates this general approach (see the `slides.qmd` files under `materials/`).

https://github.com/posit-conf-2023/quarto-r

**Reply by jensbri** [reply]
Kia ora. 

Thanks for sharing this.
We have now created a Python script and GitHub automation running as a GitHub Action to scan the repo, find qmd files, put them in a table with title, links and dates (all stemming from the yaml header of the qmd, so less worrying about keeping several instances up to date/following the dry (don't repeat yourself) principle.

Feel free to apply this action to your repo if you want.

Cheers,
Jens

**Reply by andrewpbray** [reply]
Sounds great! Have you looked into the [document listings](https://quarto.org/docs/websites/website-listings.html) feature of quarto websites? Might serve a similar purpose as your script.
